cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/DcLineSegmentConversion.java

```diff
@@ -30,32 +34,146 @@ public DcLineSegmentConversion(PropertyBag l, Context context) {
         cconverter2 = context.dc().cgmesConverterFor(iconverter2);
     }
 
+    private static double getPAc(PropertyBag p) {
+        return Double.isNaN(p.asDouble(TARGET_PPCC)) ? 0 : p.asDouble(TARGET_PPCC); // targetPpcc is the real power injection target in the AC grid in CGMES
+    }
+
+    private void updateLossFactor1(double pAC1, double poleLossP1, HvdcLine.ConvertersMode mode) {
+        if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER)) {
+            iconverter1.setLossFactor((float) (poleLossP1 / pAC1) * 100);
+        } else if (pAC1 + poleLossP1 != 0) {
+            iconverter1.setLossFactor((float) (poleLossP1 / (pAC1 + poleLossP1)) * 100);
+        }
+    }
+
+    private void updateLossFactor2(double pAC2, double poleLossP2, HvdcLine.ConvertersMode mode) {
+        if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER)) {
+            iconverter2.setLossFactor((float) (poleLossP2 / Math.abs(pAC2)) * 100);
+        } else if (Math.abs(pAC2) + poleLossP2 != 0) {
+            iconverter2.setLossFactor((float) (poleLossP2 / Math.abs(pAC2) + poleLossP2) * 100);
+        }
+    }
+
+    private void updatePowerFactor(double pAC1, double pAC2) {
+        if (iconverter1.getHvdcType() == HvdcConverterStation.HvdcType.LCC) {
+            ((LccConverterStation) iconverter1).setPowerFactor(pAC1 == 0 || Double.isNaN(iconverter1.getTerminal().getQ()) ? 0.8f : (float) Math.abs(iconverter1.getTerminal().getQ() / pAC1));
+            ((LccConverterStation) iconverter2).setPowerFactor(pAC2 == 0 || Double.isNaN(iconverter2.getTerminal().getQ()) ? 0.8f : (float) Math.abs(iconverter2.getTerminal().getQ() / pAC2));
+        }
+    }
+
+    private void updateConverterStations(double pAC1, double pAC2, double poleLossP1, double poleLossP2, HvdcLine.ConvertersMode mode) {
+
+        // update loss factors
+        if (pAC1 != 0 && pAC2 != 0) {
```

Comment:
I am not sure about the treatement of this condition. In this case, I think that we have to use :
- Only pAC1 if side 1 is rectifier and ignore pAC2 ;
- Only pAC2 if side 2 is rectifier and ignore pAC1.

Reply:
Maybe we should check if these two attributes are consistent ?

Reply:
Maybe we should check these two attributes are consistent?